application/src/main/scala/com/azavea/franklin/datamodel/Query.scala

```diff
@@ -0,0 +1,101 @@
+package com.azavea.franklin.datamodel
+
+import cats.data.NonEmptyVector
+import cats.implicits._
+import eu.timepit.refined.types.string.NonEmptyString
+import io.circe._
+import io.circe.syntax._
+
+sealed abstract class Query
+
+case class Equals(value: Json)                 extends Query
+case class NotEqualTo(value: Json)             extends Query
+case class GreaterThan(floor: Json)            extends Query
+case class GreaterThanEqual(floor: Json)       extends Query
+case class LessThan(ceiling: Json)             extends Query
+case class LessThanEqual(ceiling: Json)        extends Query
+case class StartsWith(prefix: NonEmptyString)  extends Query
+case class EndsWith(postfix: NonEmptyString)   extends Query
+case class Contains(substring: NonEmptyString) extends Query
+case class In(values: NonEmptyVector[Json])    extends Query
+
+object Query {
+
+  private def fromString(s: String, f: NonEmptyString => Query): Option[Query] =
+    NonEmptyString.from(s).toOption.map(f)
+
+  private def fromStringOrNum(js: Json, f: Json => Query): Option[Query] =
+    js.asNumber map { _ =>
+      f(js)
+    } orElse { js.asString map { _ => f(js) } }
+
+  private def errMessage(operator: String, json: Json): String =
+    s"Cannot construct `$operator` query with $json"
+
+  def queriesFromMap(unparsed: Map[String, Json]): Either[String, List[Query]] =
+    (unparsed.toList traverse {
+      case (op @ "eq", json)  => Right(Equals(json))
+      case (op @ "neq", json) => Right(NotEqualTo(json))
+      case (op @ "lt", json) =>
+        Either.fromOption(fromStringOrNum(json, LessThan.apply), errMessage(op, json))
+      case (op @ "lte", json) =>
+        Either.fromOption(
+          fromStringOrNum(json, LessThanEqual.apply),
+          errMessage(op, json)
```

Comment:
in these greater than / less than operators, it's not clear what should happen with arrays or objects. Postgres is happy to compare them but I have no idea how the ordering semantics work and I don't think anyone could propose a reasonable ordering for whether `{"class": "foo"} > {"class": "bar"}`, so I'm just rejecting the query if a comparison is applied to anything other than a number or string

Reply:
Evidence of postgres's surprising happiness:

```
franklin=# select count(1) from collection_items where item -> 'properties' -> 'class' > '3' :: jsonb;
 count 
-------
     0
(1 row)

franklin=# select count(1) from collection_items where item -> 'properties' -> 'class' > '"purple"' :: jsonb;
 count 
-------
     5
(1 row)

franklin=# select count(1) from collection_items where item -> 'properties' -> 'class' > '{"class": "purple"}' :: jsonb;
 count 
-------
     0
(1 row)
```